### DISCUSSION TITLE
Certificate expired https://netbeans.apidesign.org/maven2/

### DISCUSSION THREAD

**Discussion by racordei** [apache/netbeans#7047] [General]
Hi Dears,

Who is the responsible to renew the certificate for: https://netbeans.apidesign.org/maven2/ ????

I have an old application using netbeans RCP 82, and I can't download it due to certificate expiration

**Comment by matthiasblaesing** [comment]
@jtulach if I remember correctly netbeans.apidesign.org is your system. Could you please have a look?

**Comment by racordei** [comment]
Hi dears, any news?
Thanks in advanced.

**Comment by racordei** [comment]
Hi Dears,

Really sorry to disturb you, but anyone could help here?
I really appreciate your help.

**Comment by ursjoss** [comment]
@matthiasblaesing @jtulach Just stumbled into this. Any news if you plan to renew the certificate for  https://netbeans.apidesign.org/maven2/? thanks a bunch!

**Reply by ursjoss** [reply]
For what it's worth: The reason why I stumbled into this issue is the usage of org.netbeans.api:org-jdesktop-layout. I took the chance to migrate away from using that and therefore am not dependent on your site having a valid certificate anymore.

If the process is simple to renew the certificate (which should be the case IMHO), I would certainly vote to go for it. But if it is more elaborate, maybe an assessment might be useful whether there is actual demand for this repo being properly available.

Thanks a lot for looking into it in any case.

**Reply by jtulach** [reply]
What's your replacement for `org.netbeans.api:org-jdesktop-layout`? Isn't the necessary code already part of the JDK?

**Reply by ursjoss** [reply]
@jtulach Yes it is. The ThinkingRock code base is quite dated. My migration was this: https://github.com/thinkingrock-gtd/tr-pc/pull/154/files

**Comment by jtulach** [comment]
- I acknowledge I am seeing the the request
- keep pinging me
- better just before a weekend

**Reply by JaroslavTulach** [reply]
- looking at letsencrypt documentation
- it is different than it used to be!
   - I believe it used to be a manual process
- it talks about automatic client that re-news certificates
- I asked hosting provider for help...

**Reply by matthiasblaesing** [reply]
Yes - lets encrypt requires a client for the ACME protocol. For the "why":

Letsencrypt certificates have a lifetimeof of 90 days and might be reduced in the future. It is not feasible to renew certificates in such small timeframes. The browser forum already decided to reduce the max lifetime for certificates to 47 days.

If it helps, I use certbot, which is packaged for ubuntu and has a nice command line client (certbot) to configure the domains it shall create certificates for.

**Comment by ebarboni** [comment]
Hi @jtulach there is an ongoing issue in infra jira https://issues.apache.org/jira/browse/INFRA-19735 to maybe migrate all the old artefacts to maven/central.  Do you have metrics on this repository to have the number of artefacts / overall size of all data in this reposiroty https://netbeans.apidesign.org/maven2/
 ?

**Reply by JaroslavTulach** [reply]
$ du -hs netbeans/
14G     netbeans/

**Comment by JaroslavTulach** [comment]
Certificate for https://netbeans.apidesign.org/maven2 repository has been updated.
